src/decode.rs

```diff
@@ -7956,9 +7950,7 @@ unsafe fn decode_b(
             1 => {
                 (*(&mut *(t.l.tx_intra).as_mut_ptr().offset(by4 as isize)
                     as *mut int8_t as *mut alias8))
-                    .u8_0 = (0x1 as libc::c_int
-                    * *b_dim.offset((2 as libc::c_int + 1 as libc::c_int) as isize)
-                        as libc::c_int) as uint8_t;
+                    .u8_0 = (0x1 * b_dim[2 + 1] as libc::c_int) as uint8_t;
```

Comment:
What's the point of `0x1 * ` here?  I think the casts are redundant, going from `u8` to `c_int` to `u8` again.

```suggestion
                    .u8_0 = b_dim[2 + 1];
```

Reply:
Oh, I guess the other ones below are multiplied by different things, so this is to keep parity.  We can leave this as is then, and clean it up more later.